decorators/focus.js

```diff
@@ -2,11 +2,53 @@ var _ = require('lodash'),
   references = require('../services/references'),
   forms = require('../services/forms'),
   select = require('../services/select'),
+  sr = require('selection-range'),
   dom = require('@nymag/dom'),
   helpers = require('../services/field-helpers'),
   componentList = require('./component-list'),
   currentFocus;
 
+/**
+ * get the text offset from a click event
+ * @param {MouseEvent} e
+ * @returns {number}
+ */
+function getClickOffset(e) {
+  let range, textNode, offset, parent, parentText, parentOffset;
+
+  try {
+    if (document.caretPositionFromPoint) {
+      range = document.caretPositionFromPoint(e.clientX, e.clientY);
+      textNode = range.offsetNode;
+      offset = range.offset;
+    } else if (document.caretRangeFromPoint) {
+      range = document.caretRangeFromPoint(e.clientX, e.clientY);
+      textNode = range.startContainer;
+      offset = range.startOffset;
+    }
+
+    parent = dom.closest(textNode, '[data-uri]'); // get component el
+    parentText = parent.textContent.replace(/^(\n|.)*?Delete\s*/, ''); // remove component selector text
+    parentOffset = parentText.indexOf(textNode.textContent) + offset;
+
+    return parentOffset;
+  } catch (e) {
```

Comment:
according to the linter it's actually required for try/catch statements, even if you don't actually use it 😅